Sources/WalletConnectSign/Sign/SignClient.swift

```diff
@@ -291,15 +296,22 @@ public final class SignClient {
     // MARK: - Private
     
     private func setUpEnginesCallbacks() {
+        approveEngine.approvePublisher.sink { [unowned self] response in
+            switch response {
+            case .proposeResponse(let topic, let proposal):
+                self.sessionEngine.settlingProposal = proposal
+                self.sessionEngine.setSubscription(topic: topic)
+            case .sessionProposal(let proposal):
+                self.delegate?.didReceive(sessionProposal: proposal)
+            case .sessionRejected(proposal: let proposal, reason: let reason):
+                self.delegate?.didReject(proposal: proposal, reason: reason.publicRepresentation())
+            }
+        }.store(in: &publishers)
```

Comment:
Personally, I do not necessarily find enums publisher cleaner than callbacks 🤔 
I think some people would also say that it is less `Open–closed principle` conformant